MParT/TrainMap.h

```diff
@@ -0,0 +1,40 @@
+#ifndef MPART_TRAINMAP_H
+#define MPART_TRAINMAP_H
+
+#include <functional>
+#include <nlopt.hpp>
+#include "ConditionalMapBase.h"
+#include "MapObjective.h"
+
+namespace mpart {
+struct TrainOptions {
+    std::string opt_alg = "LD_LBFGS";
+    double opt_stopval = -std::numeric_limits<double>::infinity();
+    double opt_ftol_rel = 1e-3;
+    double opt_ftol_abs = 1e-3;
+    double opt_xtol_rel = 1e-4;
+    double opt_xtol_abs = 1e-4;
+    int opt_maxeval = 30;
+    double opt_maxtime = 100.;
```

Comment:
Do we want to set a maximum time?  I'd prefer if this was infinity (or practically so), so that the optimizer doesn't stop prematurely after a long run because I forgot to set this to something larger.